tests/Ark.Tools.ResourceWatcher.Tests/Init/TestHost.cs

```diff
@@ -0,0 +1,76 @@
+// Copyright (C) 2024 Ark Energy S.r.l. All rights reserved.
+// Licensed under the MIT License. See LICENSE file for license information. 
+using Ark.Tools.ResourceWatcher.Testing;
+using Ark.Tools.ResourceWatcher.WorkerHost;
+
+using Microsoft.Extensions.Configuration;
+
+using NodaTime;
+
+using Reqnroll;
+
+using System;
+using System.Globalization;
+using System.IO;
+
+[assembly: Microsoft.VisualStudio.TestTools.UnitTesting.DoNotParallelize]
+
+namespace Ark.Tools.ResourceWatcher.Tests.Init
+{
+    /// <summary>
+    /// Test host infrastructure for ResourceWatcher tests.
+    /// Provides access to state provider, diagnostic listener, and worker host configuration.
+    /// </summary>
+    [Binding]
+    public sealed class TestHost : IDisposable
+    {
+        private static IConfiguration? _configuration;
+
+        public static IConfiguration Configuration => _configuration ?? throw new InvalidOperationException("Configuration not initialized");
+
+        public static TestableStateProvider StateProvider { get; } = new();
```

Comment:
remove this. Move to the Context class together with creating a new WorkerHost per each Scenario and calling RunOnce().